stories/internal/Export.tsx

```diff
@@ -0,0 +1,240 @@
+const BUTTON_STYLE = { marginTop: 10, marginRight: 10 };
+
+interface IWrapperComponentProps {
+    children: (onExportReady: OnExportReady) => React.ReactChild;
+}
+
+class WrapperComponent extends React.Component<IWrapperComponentProps> {
+    private exportResult: IExportFunction;
+
+    constructor(props: IWrapperComponentProps) {
+        super(props);
+        this.onExportReady = this.onExportReady.bind(this);
+        this.exportToCSV = this.exportToCSV.bind(this);
+        this.exportToXLSX = this.exportToXLSX.bind(this);
+        this.exportWithCustomName = this.exportWithCustomName.bind(this);
+        this.exportWithMergeHeaders = this.exportWithMergeHeaders.bind(this);
+        this.doExport = this.doExport.bind(this);
+    }
+
+    public render() {
+        return (
+            <div style={WRAPPER_STYLE}>
+                <div style={INNER_STYLE}>
+                    {this.props.children(this.onExportReady)}
+                </div>
+                <div>
+                    <button style={BUTTON_STYLE} onClick={this.exportToCSV}>Export CSV</button>
+                    <button style={BUTTON_STYLE} onClick={this.exportToXLSX}>Export XLSX</button>
+                    <button style={BUTTON_STYLE} onClick={this.exportWithCustomName}>
+                        Export with custom name "CustomName"
+                    </button>
+                    <button style={BUTTON_STYLE} onClick={this.exportWithMergeHeaders}>
+                        Export with mergeHeaders
+                    </button>
+                </div>
+            </div>
+        );
+    }
+
+    private onExportReady(exportResult: IExportFunction) {
+        action('onExportReady')('fired');
+        this.exportResult = exportResult;
+    }
+
+    private exportToCSV() {
+        const options = {};
+        action('Export CSV')(options);
+        this.doExport(options);
+    }
+
+    private exportToXLSX() {
+        const options: IExportConfig = { format: 'xlsx' };
+        action('Export XLSX')(options);
+        this.doExport(options);
+    }
+
+    private exportWithCustomName() {
+        const options = { title: 'CustomName' };
+        action('Export with custom name "CustomName"')(options);
+        this.doExport(options);
+    }
+
+    private exportWithMergeHeaders() {
+        const options: IExportConfig = { format: 'xlsx', mergeHeaders: true };
+        action('Export with mergeHeaders')(options);
+        this.doExport(options);
+    }
+
+    private doExport(exportConfig: IExportConfig) {
+        this.exportResult(exportConfig)
+            .then(this.doExportSuccess, this.doExportError);
+    }
+
+    private doExportSuccess(result: IExportResponse) {
+        return action('exportResult success')(result);
+    }
+
+    private doExportError(error: Error) {
+        return action('exportResult error')(error);
+    }
+}
+
+storiesOf('Internal/Export', module)
+    .add('export chart data', () => (
+        screenshotWrap(
```

Comment:
is this screenshoting necessary? Does this screenshot verify something? Actions are important, but are these tested by screenshot? Visual design of export buttons is not production code, is it?